src/main/java/org/kuali/kfs/fp/businessobject/CapitalAssetInformationDetail.java

```diff
@@ -0,0 +1,194 @@
+        return campus;
+    }
+
+    public void setCampus(final Campus campus) {
+        this.campus = campus;
+    }
+
+    public Building getBuilding() {
+        return building;
+    }
+
+    public void setBuilding(final Building building) {
+        this.building = building;
+    }
+
+    public Room getRoom() {
+        return room;
+    }
+
+    public void setRoom(final Room room) {
+        this.room = room;
+    }
+
+    public CapitalAssetInformation getCapitalAssetInformation() {
+        return capitalAssetInformation;
+    }
+
+    public void setCapitalAssetInformation(final CapitalAssetInformation capitalAssetInformation) {
+        this.capitalAssetInformation = capitalAssetInformation;
+    }
+
+    public String getBuildingSubRoomNumber() {
+        return buildingSubRoomNumber;
+    }
+
+    public void setBuildingSubRoomNumber(final String buildingSubRoomNumber) {
+        this.buildingSubRoomNumber = buildingSubRoomNumber;
+    }
+
+    public Integer getCapitalAssetLineNumber() {
+        return capitalAssetLineNumber;
+    }
+
+    public void setCapitalAssetLineNumber(final Integer capitalAssetLineNumber) {
+        this.capitalAssetLineNumber = capitalAssetLineNumber;
+    }
+
+    /**
+     * Returns a map with the primitive field names as the key and the primitive values as the map value.
+     *
+     * @return Map a map with the primitive field names as the key and the primitive values as the map value.
+     */
+    public Map<String, Object> getValuesMap() {
+        final Map<String, Object> simpleValues = new HashMap<>();
+        simpleValues.put(KFSPropertyConstants.DOCUMENT_NUMBER, getDocumentNumber());
+        simpleValues.put(KFSPropertyConstants.CAPITAL_ASSET_LINE_NUMBER, getCapitalAssetLineNumber());
+        simpleValues.put(KFSPropertyConstants.ITEM_LINE_NUMBER, getItemLineNumber());
+        simpleValues.put(KFSPropertyConstants.CAMPUS_CODE, getCampusCode());
+        simpleValues.put(KFSPropertyConstants.BUILDING_CODE, getBuildingCode());
+        simpleValues.put(KFSPropertyConstants.BUILDING_ROOM_NUMBER, getBuildingRoomNumber());
+        return simpleValues;
+    }
+
+    // ==== CU Customization: Add handling of the extended attribute's primary key fields. ====
+
+    @Override
+    protected void beforeInsert() {
+        super.beforeInsert();
+        populatePrimaryKeyOnExtendedAttribute();
+    }
+
+    @Override
+    protected void beforeUpdate() {
+        super.beforeUpdate();
+        populatePrimaryKeyOnExtendedAttribute();
+    }
+
+    private void populatePrimaryKeyOnExtendedAttribute() {
+        CapitalAssetInformationDetailExtendedAttribute extension =
+                (CapitalAssetInformationDetailExtendedAttribute) getExtension();
+        extension.setDocumentNumber(getDocumentNumber());
+        extension.setCapitalAssetLineNumber(getCapitalAssetLineNumber());
+        extension.setItemLineNumber(getItemLineNumber());
+    }
+
+    // ==== End CU Customization ====
```

Comment:
Could this code go before the getters and setters or is our practice to put the customizations at the end of the file?

Reply:
I don't think the location within this particular file matters too much, but I'll shift them up farther for consistency with similar classes that override these methods.